Kernel/Process.cpp

```diff
@@ -607,6 +610,9 @@ void Process::finalize()
     if (veil_state() == VeilState::Dropped)
         dbgln("\x1b[01;31mProcess '{}' exited with the veil left open\x1b[0m", name());
 
+    if (g_init_pid != 0 && pid() == g_init_pid)
```

Comment:
We can't use a global `Optional` as that wants to call `__cxa_atexit`. PID 0 is always the colonel, so using 0 as a sentinel value here should be safe. Is there a better way to do this?

Reply:
I would expect pid 0 to be init_stage2, but sure. You could use a NeverDestroyed<Optional<PID>> if you like. Probably best to have it store the Distinct numeric type.